### PR TITLE
Instantiate templates

### PR DESCRIPTION
**Pull request by mschreter**

### CODE REVIEW

**Comment by peterrum** [comment]
> No it is not and takes some time :-). Would you agree if I finish in the first PR the instantiations and subsequently move the implementation to the cpp-files in problemwise PR?

Fine with me, in the case that there are no multiple instantiation errors :)

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,34 +47,60 @@ ENDIF()
...
 INCLUDE_DIRECTORIES(include ${CMAKE_CURRENT_BINARY_DIR}/include)
 
-PROJECT(MeltPoolDG)
+file(GLOB_RECURSE sources "${CMAKE_SOURCE_DIR}/include/*.hpp" "${CMAKE_SOURCE_DIR}/source/*.cpp" "${CMAKE_SOURCE_DIR}/source/**/*.cpp")
+
+ADD_LIBRARY(${libName}
+            STATIC
+            ${sources})
+
+DEAL_II_SETUP_TARGET(${libName})
```
Comment (peterrum) [inline CMakeLists.txt:84]: I am nor sure if this line is needed.

File: CMakeLists.txt [hunk 2]
```diff
@@ -33,34 +47,60 @@ ENDIF()
...
+
+################################
+# create executable
+################################
 
-SET(projectName "meltpooldg")
+SET(projectName "meltpooldg-bin")
```
Comment (peterrum) [inline CMakeLists.txt:90]: What about calling this `meltpooldg-app`?

Comment (mschreter) [reply CMakeLists.txt]: I've introduced a renaming to `meltpooldg`.

File: CMakeLists.txt (hunk 1 above)
Comment (mschreter) [reply CMakeLists.txt]: Jap, it's needed.

**Comment by mschreter** [comment]
Locally I don't have any indent problems...

**Comment by peterrum** [comment]
This is odd. For some reason it thinks that `#include <meltpooldg/utilities/utilityfunctions.hpp>` is written twice. What about `// clang-format off`. I will close my eyes in this instance^^

**Comment by mschreter** [comment]
Thanks, then this PR would be ready from my side :-).

**Comment by peterrum** [comment]
> Thanks, then this PR would be ready from my side :-).

I have removed it again ;) The problem was that clang-format is not run on the branch but on the branch merged into master. And in that version indeed the include was twice!

**Review by peterrum — APPROVED** [review]

**Pull request merged**